### PR TITLE
bugfix: request headers contains "_" should be case-insensitive matched.

### PR DESCRIPTION
**Pull request by vm-001**
I hereby granted the copyright of the changes in this pull request
to the authors of this lua-resty-core project.

Fix a bug that request header contains "_" unable to be case-insensitive matched.
e.g. 
```lua
-- Request with headers:
-- X_Foo_Header: foo

local value = ngx.req.get_headers()["X_Foo_Header"]  -- nil
```

### CODE REVIEW

**Review by zhuizhuhaomeng — APPROVED** [review]

**Comment by zhuizhuhaomeng** [comment]
@vm-001 you have removed spaces that should not be removed

**Comment by vm-001** [comment]
@zhuizhuhaomeng Ops, Looks like the IntelliJ IDEA is not intelligent at this moment. Fixed.

**Comment by vm-001** [comment]
@zhuizhuhaomeng I think the CI failure is not caused by the PR because I also found a similar failure in another empty PR(https://app.travis-ci.com/github/openresty/lua-resty-core/builds/260795767)

File: t/request.t [hunk 1]
```diff
@@ -140,6 +140,10 @@ qr/\[TRACE\s+\d+ content_by_lua\(nginx\.conf:\d+\):4 loop\]/
 
 
 === TEST 4: ngx.req.get_headers (metatable)
+--- http_config eval
+qq{
+    underscores_in_headers on;
+}
```
Comment (zhuizhuhaomeng) [inline t/request.t:146]:
```suggestion
--- http_config
    underscores_in_headers on;
```

**Review by zhuizhuhaomeng — CHANGES_REQUESTED** [review]

**Comment by vm-001** [comment]
Hi @zhuizhuhaomeng , May I know can this PR be merged?

**Comment by zhuizhuhaomeng** [comment]
@vm-001 
The CI ran failed.
Please rebase to the latest master branch code.

**Comment by zhuizhuhaomeng** [comment]
https://app.travis-ci.com/github/openresty/lua-resty-core/builds/261234448#L575
The CI failed. Please have a look at this one.

**Comment by vm-001** [comment]
> https://app.travis-ci.com/github/openresty/lua-resty-core/builds/261234448#L575 The CI failed. Please have a look at this one.

Thanks for the reminder. I've fixed the test fails. But I saw the CI still fails, have no idea what's wrong.

**Comment by vm-001** [comment]
@zhuizhuhaomeng Haven't received any reply for a while. Is it ready to merge?

File: lib/resty/core/request.lua [hunk 2]
```diff
@@ -114,7 +114,12 @@ local truncated = ffi.new("int[1]")
 
 local req_headers_mt = {
     __index = function (tb, key)
-        return rawget(tb, (str_replace_char(lower(key), '_', '-')))
+        key = lower(key)
+        local value = rawget(tb, key)
+        if value == nil then
+            value =  rawget(tb, (str_replace_char(key, '_', '-')))
```
Comment (zhuizhuhaomeng) [inline lib/resty/core/request.lua:120]:
```suggestion
            if value == nil and str_find(key, '_', 1, true) then
```

Comment (zhuizhuhaomeng) [reply lib/resty/core/request.lua:120]: this is a bit faster.

**Review by zhuizhuhaomeng — CHANGES_REQUESTED** [review]

File: lib/resty/core/request.lua [hunk 3]
```diff
@@ -114,7 +114,12 @@ local truncated = ffi.new("int[1]")
 
 local req_headers_mt = {
     __index = function (tb, key)
-        return rawget(tb, (str_replace_char(lower(key), '_', '-')))
+        key = lower(key)
+        local value = rawget(tb, key)
+        if value == nil then
+            if value == nil and str_find(key, '_', 1, true) then
+        end
```
Comment (vm-001) [inline lib/resty/core/request.lua:121]:
```suggestion
        if value == nil and str_find(key, '_', 1, true) then
            value =  rawget(tb, (str_replace_char(key, '_', '-')))
        end
```

**Comment by zhuizhuhaomeng** [comment]
This is still needed.
```
value = rawget(tb, (str_replace_char(key, '_', '-')))
```

**Comment by vm-001** [comment]
@zhuizhuhaomeng Yeah. I found the suggestion applied on the wrong line. I've made another suggestion to correct that.

**Pull request merged**